### PR TITLE
osutil: switch to -u UID:GID for strace-static

### PR DESCRIPTION
**Pull request by zyga**
This moves us off the custom patch and onto an upstream feature we've heled develop. The feature is not released yet but the patch has been integrated into the strace-static snap.

https://github.com/dilyn-corner/strace-static-snap/pull/3

Jira: SNAPDENG-19870

### CODE REVIEW

File: osutil/strace/strace.go [hunk 1]
```diff
@@ -50,8 +50,8 @@ var ExcludedSyscalls = getExcludedSyscalls()
 
 func findStrace(u *user.User) (stracePath string, userOpts []string, err error) {
 	if path := filepath.Join(dirs.SnapMountDir, "strace-static", "current", "bin", "strace"); osutil.FileExists(path) {
-		// strace-static cannot resolve usernames, pass uid/gid instead
-		return path, []string{"--uid", u.Uid, "--gid", u.Gid}, nil
+		// Strace v6.9 supports -u UID:GID which avoids the need to resolve usernames with nss.
```
Comment (Meulengracht) [inline osutil/strace/strace.go:53]: Since it required v6.9 just help me understand, do we bundle strace-static or how are we fulfilling this requirement?

Comment (Meulengracht) [reply osutil/strace/strace.go:53]: Ah right, looking more at this it needs the snap to be installed before this works?

Comment (zyga) [reply osutil/strace/strace.go:53]: There are two cases:

- you have the `strace-static` snap, just refresh or it doesn't work
- you don't have the snap - this never worked then

There are about 200-300 users of strace-static, I think it's fine to do it without any extra complexity.

**Review by Meulengracht — APPROVED** [review]

**Review by bboozzoo — APPROVED** [review]
LGTM

**Comment by zyga** [comment]
The snap is currently building. Once it reaches candidate, I will re-trigger tests.

**Comment by zyga** [comment]
@bboozzoo please re-review the last patch.

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**